SupportYourLocals.WPF/MainWindow.xaml.cs

```diff
@@ -653,6 +658,265 @@ private void DisplayErrorMessage(string message)
+            {
+                LabelUsernameError.Visibility = Visibility.Visible;
+            }
+            else if(!CheckLoginData(PasswordBox.Password, UsernameTextBox.Text))
+            {
+                LabelPasswordError.Visibility = Visibility.Visible;
+            }
+            else
+            {
+                string username = UsernameTextBox.Text;
+                int userID = csvData.GetData(username).ID;
+
+                ShowUserIDLabel.Content = "Your ID: " + userID;
+                ShowUsernameLabel.Content = username;
+
+                GridLogin.Visibility = Visibility.Collapsed; 
+                GridRegistration.Visibility = Visibility.Collapsed;
+            }
+
+            GridUserData.Visibility = Visibility.Collapsed;
+            GridUserDataToLogin.Visibility = Visibility.Collapsed;
+        }
+
+        private void RegisterButton_Clicked(object sender, RoutedEventArgs e)
+        {
+            var password = PasswordBoxR.Password;
+            var username = UsernameTextBoxR.Text;
+
+            LabelForUsedUsernameError.Visibility = Visibility.Collapsed;
+            LabelForInvalidUsernameError.Visibility = Visibility.Collapsed;
+            LabelForPasswordError.Visibility = Visibility.Collapsed;
+            LabelForConfirmPasswordError.Visibility = Visibility.Collapsed;
+            LabelForEmptyFieldsError.Visibility = Visibility.Collapsed;
+
+            if (CheckRegisterData(password, username))
+            {
+                string salt = CreateSalt(saltSize);
+                csvData.AddData(new UserData(username, GenerateHash(password, salt), salt, GenerateUserID()));
+                csvData.SaveData();
+
+                UsernameTextBoxR.Text = "";
+                PasswordBoxR.Password = "";
+                ConfirmPasswordBoxR.Password = "";
+
+                User.Visibility = Visibility.Collapsed;
+                LoginWhenRegistered.Visibility = Visibility.Visible;
+            }
+
+            GridUserData.Visibility = Visibility.Collapsed;
+            GridUserDataToLogin.Visibility = Visibility.Collapsed;
+        }
+
+        private void LoginWhenRegistered_Clicked(object sender, RoutedEventArgs e)
+        {
+            GridLogin.Visibility = Visibility.Visible;
+            GridRegistration.Visibility = Visibility.Collapsed;
+        }
+
+        private int GenerateUserID()
+        {
+            Random rnd = new Random();
+            int randomID = rnd.Next(1000, 9999);
+
+            if (Duplicates(randomID))
+            {
+                return GenerateUserID();
+            }
+            return randomID;
+        }
+
+        public bool Duplicates(int randomID)
+        {
+            foreach (var userID in csvData.GetAllData())
+            {
+                if (randomID.Equals(userID.ID))
+                {
+                    return true;
+                }
+            }
+            return false;
+        }
```

Comment:
This will not be needed once you copy the GenerateID function to the UserData class.